packages/contracts-bedrock/test/L2/L2StandardBridge.t.sol

```diff
@@ -507,6 +534,38 @@ contract L2StandardBridge_Bridge_Test is Bridge_Initializer {
         vm.expectRevert("StandardBridge: cannot send to messenger");
         l2StandardBridge.finalizeBridgeETH{ value: 100 }(alice, address(l2CrossDomainMessenger), 100, hex"");
     }
+
+    /// @dev Tests that `finalizeDeposit` reverts if a custom gas token is used.
+    function test_finalizeBridgeETH_isCustomGasToken_reverts() external {
+        vm.mockCall(
+            address(l2StandardBridge.messenger()),
+            abi.encodeWithSelector(CrossDomainMessenger.xDomainMessageSender.selector),
+            abi.encode(address(l2StandardBridge.OTHER_BRIDGE()))
+        );
+        vm.prank(address(l2CrossDomainMessenger));
+        vm.mockCall(address(l1Block), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2)));
+        vm.expectRevert("StandardBridge: cannot bridge ETH with custom gas token");
+
+        l2StandardBridge.finalizeBridgeETH(alice, alice, 100, hex"");
+    }
+
+    /// @dev Tests that bridging ETH reverts if a custom gas token is used.
+    function test_bridgeETHTo_isCustomGasToken_reverts() external {
+        vm.prank(alice, alice);
+        vm.mockCall(address(l1Block), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2)));
+        vm.expectRevert("StandardBridge: cannot bridge ETH with custom gas token");
+
+        l2StandardBridge.bridgeETHTo(bob, 60000, hex"dead");
+    }
+
+    /// @dev Tests that bridging ETH reverts if a custom gas token is used.
+    function test_bridgeETH_isCustomGasToken_reverts() external {
+        vm.prank(alice, alice);
+        vm.mockCall(address(l1Block), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(1), uint8(2)));
+        vm.expectRevert("StandardBridge: cannot bridge ETH with custom gas token");
+
+        l2StandardBridge.bridgeETH(50000, hex"dead");
+    }
```

Comment:
This needs `receive` function coverage, similar to L1StandardBridge

Reply:
There are also a bunch of legacy functions on both L1 and L2 standard bridges noted in the spec that should revert